src/decompression.jl

```diff
@@ -264,12 +263,11 @@ end
 
 ## ColumnColoringResult
 
-function decompress!(
-    A::AbstractMatrix{R}, B::AbstractMatrix{R}, result::ColumnColoringResult
-) where {R<:Real}
-    @compat (; S, color) = result
+function decompress!(A::AbstractMatrix, B::AbstractMatrix, result::ColumnColoringResult)
+    @compat (; color) = result
+    S = result.bg.S2
     check_same_pattern(A, S)
-    A .= zero(R)
+    fill!(A, zero(eltype(A)))
```

Comment:
I recently discovered that `fill!(A, val)` is much more efficient than `A .= val`

Reply:
On devrait rzjouter un petit check avec le promote_type pour être certains que ça marche.

Reply:
En vrai si ça marche pas Julia nous lancera une exception sur la promotion et ce sera tout aussi clair qu'un message d'erreur manuel non ?

Reply:
Pas faux. On peut laisser comme ça